main/src/main/java/cgeo/geocaching/CacheDetailActivity.java

```diff
@@ -1847,6 +1848,46 @@ public void setContent() {
 
         }
 
+        private void translateListing() {
+            final CacheDetailActivity cda = (CacheDetailActivity) getActivity();
+
+            if (cda.translationStatus.isTranslated()) {
+                cda.translationStatus.setNotTranslated();
+                reloadDescription(cda, cache, true, 0, cda.descriptionStyle);
+                if (TextUtils.containsHtml(cache.getHint())) {
+                    binding.hint.setText(HtmlCompat.fromHtml(cache.getHint(), HtmlCompat.FROM_HTML_MODE_LEGACY, new HtmlImage(cache.getGeocode(), false, false, false), null), TextView.BufferType.SPANNABLE);
+                } else {
+                    binding.hint.setText(cache.getHint());
+                }
+                return;
+            }
+
+            final OfflineTranslateUtils.Language sourceLng = cda.translationStatus.getSourceLanguage();
+            cda.translationStatus.startTranslation(2, cda, cda.findViewById(R.id.description_translate_button));
+
+            OfflineTranslateUtils.getTranslator(cda, sourceLng,
```

Comment:
IMHO, the callback from Status should already pass the fully initialized Translator object. I will comment more on this on the Status object.